packages/pyright-internal/src/parser/parser.ts

```diff
@@ -1207,10 +1207,17 @@ export class Parser {
     private _parseSuite(isFunction = false, skipBody = false, postColonCallback?: () => void): SuiteNode {
         const nextToken = this._peekToken();
         const suite = SuiteNode.create(nextToken);
+        let comparisonOperator: OperatorType | undefined;
 
         if (!this._consumeTokenIfType(TokenType.Colon)) {
-            this._addError(Localizer.Diagnostic.expectedColon(), nextToken);
-
+            if (nextToken.type === TokenType.Operator) {
+                comparisonOperator = this._peekOperatorType();
+                if (comparisonOperator === OperatorType.Assign) {
```

Comment:
Can we just inline this._peekOperatorType() and remove the comparisonOperator var?

There are some tests in parser.test.ts that could be extended with a test to cover this with an example like:

```
a = 0
if a = 1:
    pass
```

Reply:
Inline code - [`45d87ec`](https://github.com/microbit-foundation/pyright/pull/10/commits/45d87ecaeb9f1945275662bd0f7cd4bb42ecbea6).
Add tests - [`28a8cff`](https://github.com/microbit-foundation/pyright/pull/10/commits/28a8cffcaecc4d402204dd6a88044a52787a2525).